cmake/SetupInstall.cmake

```diff
@@ -0,0 +1,40 @@
+
+ # install
+ install(TARGETS ${PROJECT_NAME}
+    EXPORT ${PROJECT_NAME}_Targets
+    ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
+    LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
+    RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
+)
```

Comment:
What are the differences between all these?

Reply:
This is just CMake boilerplate, e.g. when installed, it will go in `$PREFIX/bin`, `$PREFIX/lib`, ect